test/integration_test.go

```diff
@@ -619,6 +619,26 @@ func (ts *IntegrationTestSuite) TestActivityPause() {
 	ts.True(desc.GetPendingActivities()[0].GetPaused())
 }
 
+func (ts *IntegrationTestSuite) TestActivityReset() {
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+	// Run ActivityHeartbeat workflow, this workflow will call
+	// ActivityToBeReset activity twice, the first call will test resetting an activity successfully
+	// and the second call will test completing the activity after it has been reset
+	run, err := ts.client.ExecuteWorkflow(ctx,
+		ts.startWorkflowOptions("test-activity-reset"), ts.workflows.ActivityHeartbeatReset)
+	ts.NoError(err)
+	// Wait for the workflow to finish
+	var result string
+	err = run.Get(ctx, &result)
+	ts.NoError(err)
+	// Check the result - ensure ErrActivityReset was thrown, heartbeat details were reset, and num attempts were reset
```

Comment:
Does this actually check the details were reset? Maybe I am misreading the test, but it looks like the activity only runs once no?

Reply:
It should, activity reset/pause is propagated via heartbeat response. I think running the activity once should be enough to do:

`Reset` -> `Heartbeat` (propagate the reset response, transform + emit an activity task cancellation) -> `Receive cancellation in activity (i.e. ErrActivityReset)`

Which is enough to check the reset flow, unless I'm missing something